crates/bevy_ecs/src/world/component_constants.rs

```diff
@@ -16,36 +16,39 @@ pub const ON_REMOVE: ComponentId = ComponentId::new(3);
 /// [`ComponentId`] for [`OnDespawn`]
 pub const ON_DESPAWN: ComponentId = ComponentId::new(4);
 
-/// Trigger emitted when a component is added to an entity. See [`crate::component::ComponentHooks::on_add`]
-/// for more information.
+/// Trigger emitted when a component is inserted onto an entity without it. Runs before `OnInsert`.
+/// See [`crate::component::ComponentHooks::on_add`] for more information.
 #[derive(Event, Debug)]
 #[cfg_attr(feature = "bevy_reflect", derive(Reflect))]
 #[cfg_attr(feature = "bevy_reflect", reflect(Debug))]
 pub struct OnAdd;
 
-/// Trigger emitted when a component is inserted onto an entity. See [`crate::component::ComponentHooks::on_insert`]
-/// for more information.
+/// Trigger emitted when a component is inserted onto an entity, regardless of whether or not it
+/// already had it. Runs after `OnAdd`, if it ran.
+/// See [`crate::component::ComponentHooks::on_insert`] for more information.
 #[derive(Event, Debug)]
 #[cfg_attr(feature = "bevy_reflect", derive(Reflect))]
 #[cfg_attr(feature = "bevy_reflect", reflect(Debug))]
 pub struct OnInsert;
 
-/// Trigger emitted when a component is replaced on an entity. See [`crate::component::ComponentHooks::on_replace`]
-/// for more information.
+/// Trigger emitted when a component is inserted onto an entity that already had it, and runs before
```

Comment:
"is inserted onto an entity that already had it" feels like it might be less clear than the previous "replaced"
Maybe "replaced on an entity by another component of the same type"? Feels a bit long but it's clearer.